service/matching/taskListManager.go

```diff
@@ -323,6 +405,12 @@ func (c *taskListManagerImpl) getTask(ctx context.Context) (*getTaskResult, erro
 		}()
 	}
 
+	// Wait till long poll expiration for token. If token acquired, proceed.
+	if ok := c.rateLimiter.Consume(1, c.config.LongPollExpirationInterval); !ok {
```

Comment:
the caller context deadline could technically be less than the LongPollExpirationInterval, this probably needs to be fixed later.

Reply:
Good catch, fixing.